pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -98,35 +99,32 @@ func (p *PoolManager) AllocateVirtualMachineMac(virtualMachine *kubevirt.Virtual
 	return nil
 }
 
-func (p *PoolManager) ReleaseVirtualMachineMac(vm *kubevirt.VirtualMachine) error {
+func (p *PoolManager) ReleaseVirtualMachineMac(vm *kubevirt.VirtualMachine, reconcileLog logr.Logger) error {
```

Comment:
Hey what do you think about removing log from here and creating the following at PoolManager
```golang
func (p *PoolManager) String() string {
...
}
```

This way you can do something like the following

```golang
fmt.Sprintf("KMP pool manager: %s", poolManager)
```

Reply:
How does it help with the identification of single execution?

Reply:
Not much, just passing the logger to all functions as an specific case of `sillver bullet` feels uncomplete, maybe we need to pass some kind of requestContext to all the calls here so we can really trace things.

Reply:
I hate the appeal of dependency injection (countless arguments passed down the stack everywhere), but what is better context than single purpose obejct (logger) doing one thing, keeping all the information while allowing its users to make transparent logs without understanding that they are part of Reconcile loop

Reply:
ack, fair enough.